services/service.go

```diff
@@ -274,17 +279,32 @@ func NewService(
 // getIterationsForLimitQuery is used to get iterations of querying data from the contract with given rpc limit for blocks
 // and latest block number. Limit by default (if given limit is 0) is set to 20 000 blocks
 func (s *Service) getIterationsForLimitQuery(limit uint64) (iterations uint64, lastBlock uint64, err error) {
-	lastBlock, err = s.rpcClient.BlockNumber(context.Background())
-	if err != nil {
-		logger.Log().WithField("layer", "Service-getIterationsForLimitQuery").Errorf("get latest block rpc error: %v", err.Error())
-		return 0, 0, errors.GetRPCProviderErr(err, "BlockNumber")
+	return s.getIterationsForQuery(0, 0, limit)
+}
+
+func (s *Service) getIterationsForQuery(fromBlock uint64, toBlock uint64, limit uint64) (iterations uint64, lastBlock uint64, err error) {
+	if fromBlock == 0 {
+		fromBlock = s.perpsMarketFirstBlock
+	}
+
+	if toBlock == 0 {
+		lastBlock, err = s.rpcClient.BlockNumber(context.Background())
```

Comment:
if toBlock != 0, last block will be 0?

Reply:
its a typo, fixed in the last commit. Thx for spotting!